crates/rpc/rpc/src/eth/pubsub.rs

```diff
@@ -106,6 +109,58 @@ async fn handle_accepted<Client, Pool, Events>(
         }
         SubscriptionKind::Syncing => {
             // TODO subscribe new blocks -> read is_syncing from network
+            tokio::spawn(async move {
+                let mut block_subscription = pubsub.chain_events.subscribe_new_blocks();
+                let initial_sync_status = pubsub.client.is_syncing();
+
+                let initial_sub_res = if initial_sync_status {
```

Comment:
we need this again for OnNewBlock, so this should probably be a function or closure so we can reuse